### PR TITLE
First crack at a clowdapp template for drift backend

### PR DESCRIPTION
**Pull request by msager27**
## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: clowdapp.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    scaleTargetRef:
+      apiVersion: v1
+      kind: DeploymentConfig
+      name: drift-backend
+    minReplicas: ${{MIN_REPLICAS}}
+    maxReplicas: ${{MAX_REPLICAS}}
+    targetCPUUtilizationPercentage: 50
+
+parameters:
+  - name: HOST_INVENTORY_HOST
```
Comment (Victoremepunto) [inline clowdapp.yaml:130]: The dependent services like inventory will be provided by clowder and need to be set at runtime reading the values using the Clowder config helper lib

Comment (Victoremepunto) [reply clowdapp.yaml:130]: the same goes for rbac , etc.

File: clowdapp.yaml [hunk 2]
```diff
@@ -0,0 +1,167 @@
+---
+apiVersion: v1
+kind: Template
+metadata:
+  name: drift-backend
+
+objects:
```
Comment (Victoremepunto) [inline clowdapp.yaml:7]: need to add a dependencies with the list of services your application depends on. See an example here: 

https://github.com/RedHatInsights/compliance-backend/pull/866/commits/085c3fcafdfd1509d0e8ef74e3c3208a594edf08#diff-0f518b5985b4ea86412d39c9757171411345db257cb061b84bb314a7a2b3d5e9R27

and for reference: https://redhatinsights.github.io/clowder/clowder/dev/api_reference.html

search for dependencies, there’s hard and optional dependencies 

the hard ones, means the deployment will wait for those services to be available 
the optional will not make your deployment to wait. however, once the optional dependencies are ready, your pods will be recreated to reload the configuration of the new service.

**Review by Victoremepunto — CHANGES_REQUESTED** [review]
Looks good, left some comments

File: clowdapp.yaml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+          - name: CW_AWS_ACCESS_KEY_ID
+            valueFrom:
+              secretKeyRef:
+                key: CW_AWS_ACCESS_KEY_ID
+                name: cloudwatch
+          - name: CW_AWS_SECRET_ACCESS_KEY
+            valueFrom:
+              secretKeyRef:
+                key: CW_AWS_SECRET_ACCESS_KEY
+                name: cloudwatch
```
Comment (Victoremepunto) [inline clowdapp.yaml:87]: Cloudwatch config is also set by Clowder, so you can also drop this section . Clowder should provide the `access key`, the `region`, the `secret access key` and the `log group`.

File: clowdapp.yaml (hunk 2 above)
Comment (msager27) [reply clowdapp.yaml:7]: Thanks for reviewing, Victor.  I've made some updates.  I'm sure it'll need further fixes/refining as we move forward.

**Pull request merged**